sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -170,12 +174,81 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
     assessmentId: number,
     status: AssessmentStatus,
   ): Promise<UpdateResult> {
-    return this.repo.update(
+    return await this.dataSource.transaction(
+      async (transactionalEntityManager) => {
+        const updateResult = await transactionalEntityManager
+          .getRepository(StudentAssessment)
+          .createQueryBuilder()
+          .update(StudentAssessment)
+          .set({
+            noaApprovalStatus: status,
+          })
+          .where("id = :assessmentId", { assessmentId })
+          .andWhere("noaApprovalStatus is NULL")
+          .execute();
+
+        if (updateResult.affected !== 1) {
```

Comment:
We cannot throw this type of error here because of this condition `andWhere("noaApprovalStatus is NULL")`.
We must ensure the idempotency of the worker's methods. If this method is called twice it must produce the same result.
If an exception will be thrown due to an `Assessment not found.` we must ensure that it is because the id was not present.
Also, if the method is called twice we must ensure that the notification is not sent twice.